### PR TITLE
Bug fix firewall policy association

### LINKED ISSUE #11: Bug: Azure Firewall deployment does not configure specified Firewall Polices
## Summary
When attempting to configure the module deployed Azure Firewall with a specified Firewall Policy the policy is not deployed to the firewall. Looking into the module code it looks like the `firewall_policy_id` argument is missing from the firewall resource.

## Reproduction
```hcl
"aue-vhub-fw" = {
      sku_name         = "AZFW_Hub"
      sku_tier         = "Standard"
      name             = "aue-hub-fw"
      virtual_hub_name = "aue-vhub"
      firewall_policy_id = "/subscriptions/***/resourceGroups/tvmVwanRg/providers/Microsoft.Network/firewallPolicies/afwp-tvmhub-aue-001"
    }
```
## Remediation
```hcl
resource "azurerm_firewall" "fw" {
  for_each = var.firewalls

  name                = each.value.name
  location            = azurerm_virtual_hub.virtual_hub[each.value.virtual_hub_name].location
  resource_group_name = azurerm_virtual_hub.virtual_hub[each.value.virtual_hub_name].resource_group_name
  sku_name            = each.value.sku_name
  sku_tier            = each.value.sku_tier
  firewall_policy_id  = each.value.firewall_policy_id
  tags                = try(each.value.tags, {})

  virtual_hub {
    virtual_hub_id  = azurerm_virtual_hub.virtual_hub[each.value.virtual_hub_name].id
  }
}

```

### PR DESCRIPTION
**Pull request by pagyP**
## Type of Change

- [ ] Non-module change (e.g. CI/CD, documentation, etc.)
- [x] Azure Verified Module updates:
  - [x] Bugfix containing backwards compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `locals.version.tf.json`:
    - [x] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `locals.version.tf.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `locals.version.tf.json`.
  - [ ] Update to documentation

# Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings
- [x] I did run all  [pre-commit](https://azure.github.io/Azure-Verified-Modules/contributing/terraform/terraform-contribution-flow/#5-run-pre-commit-checks) checks

### CODE REVIEW

File: examples/firewall-linked-firewall-policy/.terraform-docs.yml
Comment (matt-FFFFFF) [inline examples/firewall-linked-firewall-policy/.terraform-docs.yml:1]: This file does not need to be here

**Review by matt-FFFFFF — COMMENTED** [review]
One minor comment. Otherwise lgtm

File: examples/firewall-linked-firewall-policy/.terraform-docs.yml
Comment (pagyP) [reply examples/firewall-linked-firewall-policy/.terraform-docs.yml:1]: ooops - I've removed that file.

**Review by khushal08 — APPROVED** [review]
LGTM

**Pull request merged**